pandas/io/pickle.py

```diff
@@ -146,17 +146,26 @@ def read_pickle(path, compression="infer"):
 
     # 1) try standard libary Pickle
     # 2) try pickle_compat (older pandas version) to handle subclass changes
-    # 3) try pickle_compat with latin1 encoding
+
+    excs_to_catch = (TypeError, AttributeError, ImportError)
+    if PY36:
+        excs_to_catch += (ModuleNotFoundError,)
 
     try:
         with warnings.catch_warnings(record=True):
             # We want to silence any warnings about, e.g. moved modules.
             warnings.simplefilter("ignore", Warning)
             return pickle.load(f)
-    except Exception:
+    except excs_to_catch:
+        # e.g.
+        #  "No module named 'pandas.core.sparse.series'"
+        #  "Can't get attribute '__nat_unpickle' on <module 'pandas._libs.tslib"
+        #  TypeError: _reconstruct: First argument must be a sub-type of ndarray
+        #    --> appears specific to pytables in 0.8.0 whatsnew, see GH#28645
         try:
             return pc.load(f, encoding=None)
-        except Exception:
+        except UnicodeDecodeError:
```

Comment:
can't remove that code in 0.8.0 whatsnew? (e.g. make it a code-block)?

Reply:
looks like this worked